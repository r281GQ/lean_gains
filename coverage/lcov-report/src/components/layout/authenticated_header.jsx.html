<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/authenticated_header.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> authenticated_header.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Toolbar } from 'material-ui';
&nbsp;
import RouterNavigation from './router_navigation';
import Main from './main';
import Exercises from './exercises';
import User from './user';
import Macros from './macros';
import SaveIndicator from './save_indicator';
&nbsp;
&nbsp;
&nbsp;
const AuthenticatedHeader = <span class="fstat-no" title="function not covered" >({</span>
  openSideBar,
  exercises,
  isFetching,
  isLoading,
  userName,
  photo,
  todaysMacros,
  goBack,
  goForward
}) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;Toolbar style={{ backgroundColor: '#757575', color: '#EEEEEE' }}&gt;</span>
    &lt;Main openSideBar={openSideBar} /&gt;
    &lt;RouterNavigation goBack={goBack} goForward={goForward} /&gt;
    &lt;Exercises exercises={exercises} /&gt;
    &lt;Macros todaysMacros={todaysMacros} isFetching={isFetching} /&gt;
    &lt;SaveIndicator isLoading={isLoading} /&gt;
    &lt;User userName={userName} photo={photo} /&gt;
  &lt;/Toolbar&gt;;
&nbsp;
AuthenticatedHeader.propTypes = {
  openSideBar: PropTypes.func,
  exercises: PropTypes.arrayOf(PropTypes.string),
  isFetching: PropTypes.bool,
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  todaysMacros: PropTypes.object
};
&nbsp;
export default AuthenticatedHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
