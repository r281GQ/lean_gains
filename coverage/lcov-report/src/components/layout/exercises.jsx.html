<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/exercises.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> exercises.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import * as _ from 'lodash';
import {
  IconButton,
  ToolbarGroup,
  ToolbarTitle,
  MenuItem,
  IconMenu
} from 'material-ui';
import ContentFilter from 'material-ui/svg-icons/navigation/arrow-downward';
&nbsp;
const Exercises = <span class="fstat-no" title="function not covered" >({</span> exercises }) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;ToolbarGroup&gt;</span>
    &lt;ToolbarTitle style={{ color: '#EEEEEE' }} text="Exercies for today" /&gt;
    &lt;IconMenu
      value="0"
      iconButtonElement={
        &lt;IconButton&gt;
          &lt;ContentFilter color="white" /&gt;
        &lt;/IconButton&gt;
      }
    &gt;
      {!_.isEmpty(exercises)
        ? _.map(exercises, <span class="fstat-no" title="function not covered" >ex</span>ec =&gt;
<span class="cstat-no" title="statement not covered" >            &lt;MenuItem</span>
              key={exercises.indexOf(exec)}
              value={exercises.indexOf(exec)}
              primaryText={exec.toUpperCase()}
            /&gt;
          )
        : &lt;MenuItem key={0} value={0} primaryText="NO EXERCISE FOR TODAY" /&gt;}
    &lt;/IconMenu&gt;
  &lt;/ToolbarGroup&gt;;
&nbsp;
Exercises.propTypes = {
  exercises: PropTypes.arrayOf(PropTypes.string)
};
&nbsp;
export default Exercises;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
