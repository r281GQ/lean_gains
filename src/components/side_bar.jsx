import React from 'react';
import { Link } from 'react-router-dom';
import { Drawer, MenuItem, FlatButton, Divider, Menu } from 'material-ui';
import styled from 'styled-components';

import { closeSideBar } from './../store/actionCreators/app_action_creators';
import { logOut } from './../store/actionCreators/auth_action_creators';

const LinkWithoutUnderLine = styled(Link)`
  text-decoration: none;
`;

const SideBarComponent = () => {
  return (
    <Drawer
      open={isSideBarOpen}
      docked={false}
      onRequestChange={() => closeSideBar()}
    >
      <Menu desktop={true}>
        <MenuItem onClick={() => closeSideBar()} primaryText="Close" />

        <LinkWithoutUnderLine
          to="/app/workoutlogs"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Workout logs" />
        </LinkWithoutUnderLine>
        <LinkWithoutUnderLine
          to="/app/kcaltarget"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Set calorie target" />
        </LinkWithoutUnderLine>
        <LinkWithoutUnderLine
          to="/app/dailylogs"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Daily logs" />
        </LinkWithoutUnderLine>

        <LinkWithoutUnderLine
          to="/app/workouttargets"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Set workout target" />
        </LinkWithoutUnderLine>
        <LinkWithoutUnderLine
          to="/app/userdetails"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Change user details" />
        </LinkWithoutUnderLine>
        <LinkWithoutUnderLine
          to="/app/kcaltracker"
          onClick={() => closeSideBar()}
        >
          <MenuItem primaryText="Track calories" />
        </LinkWithoutUnderLine>
        <Divider />
        <MenuItem onClick={() => logOut()} primaryText="Log out" />
      </Menu>
    </Drawer>
  );
};

export default SideBarComponent;
