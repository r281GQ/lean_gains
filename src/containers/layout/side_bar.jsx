 import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Drawer, MenuItem, Divider, FontIcon } from 'material-ui';
import PropTypes from 'prop-types';
import { closeSideBar } from './../../store/actionCreators/app_action_creators';
import { logOut } from './../../store/actionCreators/auth_action_creators';

import styled from 'styled-components';
import ArrowBack from 'material-ui/svg-icons/navigation/arrow-back';

const LinkWithoutUnderLine = styled(Link)`
  text-decoration: none;
`;

//TODO: implement help sidebar
const SideBarContainer = ({ isSideBarOpen, closeSideBar, logOut, links }) => {
  return (
    <Drawer
      open={isSideBarOpen}
      docked={false}
      onRequestChange={() => closeSideBar()}
    >
      <MenuItem
        onClick={() => closeSideBar()}
        primaryText="Close"
        rightIcon={<ArrowBack />}
      />
      <Divider />
      <LinkWithoutUnderLine
        to="/app/kcaltracker"
        onClick={() => closeSideBar()}
      >
        <MenuItem
          primaryText="Track calories"
          rightIcon={<FontIcon className="fa fa-pencil-square-o" />}
        />
      </LinkWithoutUnderLine>
      <LinkWithoutUnderLine
        to="/app/workoutlogs"
        onClick={() => closeSideBar()}
      >
        <MenuItem
          primaryText="Workout logs"
          rightIcon={<FontIcon className="fa fa-pencil" />}
        />
      </LinkWithoutUnderLine>
      <LinkWithoutUnderLine to="/app/dailylogs" onClick={() => closeSideBar()}>
        <MenuItem
          primaryText="Daily logs"
          rightIcon={<FontIcon className="fa fa-pencil-square" />}
        />
      </LinkWithoutUnderLine>
      <LinkWithoutUnderLine to="/app/kcaltarget" onClick={() => closeSideBar()}>
        <MenuItem
          primaryText="Set calorie target"
          rightIcon={<FontIcon className="fa fa-dot-circle-o" />}
        />
      </LinkWithoutUnderLine>

      <LinkWithoutUnderLine
        to="/app/workouttargets"
        onClick={() => closeSideBar()}
      >
        <MenuItem
          primaryText="Set workout target"
          rightIcon={<FontIcon className="fa fa-bullseye" />}
        />
      </LinkWithoutUnderLine>

      <LinkWithoutUnderLine
        to="/app/userdetails"
        onClick={() => closeSideBar()}
      >
        <MenuItem
          primaryText="Change user details"
          rightIcon={<FontIcon className="fa fa-user" />}
        />
      </LinkWithoutUnderLine>
      <Divider />
      <LinkWithoutUnderLine to="/app/about" onClick={() => closeSideBar()}>
        <MenuItem
          primaryText="About"
          rightIcon={<FontIcon className="fa fa-info" />}
        />
      </LinkWithoutUnderLine>

      <Divider />
      <MenuItem
        onClick={() => logOut()}
        primaryText="Log out"
        rightIcon={<FontIcon className="fa fa-sign-out" />}
      />
    </Drawer>
  );
};

const mapStateToProps = state => ({
  isSideBarOpen: state.getIn(['app', 'isSideBarOpen'])
});

export default connect(mapStateToProps, { closeSideBar, logOut })(
  SideBarContainer
);
