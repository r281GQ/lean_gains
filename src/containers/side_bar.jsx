import React from "react";

import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Drawer, MenuItem, FlatButton } from "material-ui";

import { closeSideBar } from "./../store/actionCreators/app_action_creators";

const SideBarContainer = ({ isSideBarOpen, closeSideBar }) => {
  return (
    <Drawer
      open={isSideBarOpen}
      docked={false}
      onRequestChange={() => closeSideBar()}
    >
      <MenuItem onClick={() => closeSideBar()}>Close</MenuItem>

      <Link
        to="/workoutlogs"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>Workout logs</MenuItem>
      </Link>
      <Link
        to="/KcalLog"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>kolrialog</MenuItem>
      </Link>
      <Link
        to="/dailylogs"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>
          <span>Daily logs</span>
        </MenuItem>
      </Link>

      <Link
        to="/workouttargets"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>
          {" "}<span>workouttarget</span>
        </MenuItem>
      </Link>
      <Link
        to="/userdetails"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>
          {" "}<span>userdetails</span>
        </MenuItem>
      </Link>
      <Link
        to="/kcaltracker"
        style={{ textDecoration: "none" }}
        onClick={() => closeSideBar()}
      >
        <MenuItem>
          {" "}<span>kl</span>
        </MenuItem>
      </Link>
    </Drawer>
  );
};

const mapStateToProps = state => ({
  isSideBarOpen: state.getIn(["app", "isSideBarOpen"])
});

const mapDispatchToProps = dispatch => ({
  closeSideBar: () => dispatch(closeSideBar())
});

export default connect(mapStateToProps, mapDispatchToProps)(SideBarContainer);
