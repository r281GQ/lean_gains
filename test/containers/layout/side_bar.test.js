import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { PureSideBarContainer } from './../../../src/containers/layout/side_bar';

describe('SideBar test', () => {
  it('should render', () => {
    const logOutSpy = sinon.spy();
    const closeSideBarSpy = sinon.spy();
    const component = shallow(
      <PureSideBarContainer logOut={logOutSpy} closeSideBar={closeSideBarSpy} />
    );

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should call logOutSpy & closeSideBarSpy', () => {
    const logOutSpy = sinon.spy();
    const closeSideBarSpy = sinon.spy();
    const component = shallow(
      <PureSideBarContainer logOut={logOutSpy} closeSideBar={closeSideBarSpy} />
    );

    component.instance()._handleLogOut();
    component.instance()._handleCloseSideBar();

    expect(closeSideBarSpy.calledOnce).toBe(true);
    expect(logOutSpy.calledOnce).toBe(true);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
