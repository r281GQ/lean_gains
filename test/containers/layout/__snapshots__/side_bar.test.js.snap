// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar test should call logOutSpy & closeSideBarSpy 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={null}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
    primaryText="Close"
    rightIcon={<NavigationArrowBack />}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
  <Divider
    inset={false}
  />
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/kcaltracker"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Track calories"
      rightIcon={
        <FontIcon
          className="fa fa-pencil-square-o"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/workoutlogs"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Workout logs"
      rightIcon={
        <FontIcon
          className="fa fa-pencil"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/dailylogs"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Daily logs"
      rightIcon={
        <FontIcon
          className="fa fa-pencil-square"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/kcaltarget"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Set calorie target"
      rightIcon={
        <FontIcon
          className="fa fa-dot-circle-o"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/workouttargets"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Set workout target"
      rightIcon={
        <FontIcon
          className="fa fa-bullseye"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/userdetails"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Change user details"
      rightIcon={
        <FontIcon
          className="fa fa-user"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Divider
    inset={false}
  />
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/about"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="About"
      rightIcon={
        <FontIcon
          className="fa fa-info"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Divider
    inset={false}
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
    primaryText="Log out"
    rightIcon={
      <FontIcon
        className="fa fa-sign-out"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    }
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
</Drawer>
`;

exports[`SideBar test should render 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={null}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
    primaryText="Close"
    rightIcon={<NavigationArrowBack />}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
  <Divider
    inset={false}
  />
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/kcaltracker"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Track calories"
      rightIcon={
        <FontIcon
          className="fa fa-pencil-square-o"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/workoutlogs"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Workout logs"
      rightIcon={
        <FontIcon
          className="fa fa-pencil"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/dailylogs"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Daily logs"
      rightIcon={
        <FontIcon
          className="fa fa-pencil-square"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/kcaltarget"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Set calorie target"
      rightIcon={
        <FontIcon
          className="fa fa-dot-circle-o"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/workouttargets"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Set workout target"
      rightIcon={
        <FontIcon
          className="fa fa-bullseye"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/userdetails"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Change user details"
      rightIcon={
        <FontIcon
          className="fa fa-user"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Divider
    inset={false}
  />
  <Link
    className="link-no-underline"
    onClick={[Function]}
    replace={false}
    to="/app/about"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="About"
      rightIcon={
        <FontIcon
          className="fa fa-info"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      }
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Link>
  <Divider
    inset={false}
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
    primaryText="Log out"
    rightIcon={
      <FontIcon
        className="fa fa-sign-out"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    }
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
</Drawer>
`;
